src/symbol/placement.js

```diff
@@ -350,29 +416,66 @@ export class Placement {
                         }
                     }
 
-                    for (const anchor of anchors) {
-                        placedGlyphBoxes = this.attemptAnchorPlacement(
-                            anchor, textBox, width, height, symbolInstance.radialTextOffset,
-                            textBoxScale, rotateWithMap, pitchWithMap, textPixelRatio, posMatrix,
-                            collisionGroup, textAllowOverlap, symbolInstance, bucket);
-                        if (placedGlyphBoxes) {
-                            placeText = true;
-                            break;
+                    const placeBoxForVariableAnchors = (collisionTextBox, orientation) => {
+                        const width = collisionTextBox.x2 - collisionTextBox.x1;
+                        const height = collisionTextBox.y2 - collisionTextBox.y1;
+                        const textBoxScale = symbolInstance.textBoxScale;
+
+                        let placedBox = { box: null, offscreen: null };
```

Comment:
The `placedBox` type is Missing flow type annotation. Probably also good to annotate the return type of `attemptAnchorPlacement()`

Reply:
For my own learning as I am still new to Flow:
- how would this pass Flow tests if some annotation was missing? 
- why would we want to annotate `placedBox` but not `width` `height` etc?
- If I'm reading correctly, many (most?) of the functions in this module - [including `attemptAnchorPlacement()`](https://github.com/mapbox/mapbox-gl-js/blob/master/src/symbol/placement.js#L238) - were already missing return value annotations. Should they all be added, and if so does it make sense for that to be part of this PR? Seems like it might make more sense to be addressed separately..

Reply:
> how would this pass Flow tests if some annotation was missing?

Flow does not test for missing annotations and cannot test objects/methods that are not annotated. That allows subtle bugs in those areas of code to be introduced without the ability to 

> why would we want to annotate placedBox but not width height etc?

`collisionTextBox` is typed as a `SingleCollisionBox`, implying that `width` and `height` will be treated as `number` types.

> If I'm reading correctly, many (most?) of the functions in this module [...] were already missing return value annotations. Should they all be added, and if so does it make sense for that to be part of this PR? Seems like it might make more sense to be addressed separately..

I prefer to add annotations to methods when I make a bunch of changes to them - it allows for learning the contracts of a method, and how it is used. But this can also be tackled in a separate PR - as a larger fix.